internal/controller/providers/pxc/applier.go

```diff
@@ -117,6 +143,39 @@ func (p *applier) Engine() error {
 	}
 
 	pxc.Spec.SecretsName = p.DB.Spec.Engine.UserSecretsName
+	// XXX: PXCO v1.17.0 has a bug in the auto generation of the secrets where
+	// there's a 1.5% chance that the proxyadmin password starts with the '*'
+	// character, which will be rejected by the secret validation, leaving the
+	// cluster stuck in the 'initializing' state.
+	// See: https://perconadev.atlassian.net/browse/K8SPXC-1568
+	// To work around this, we need to set the password ourselves to bypass the
+	// buggy auto generation.
+	if p.DBEngine.Status.OperatorVersion == "1.17.0" {
+		secret := &corev1.Secret{}
+		err := p.C.Get(p.ctx, types.NamespacedName{
+			Name:      p.DB.Spec.Engine.UserSecretsName,
+			Namespace: p.DB.GetNamespace(),
+		}, secret)
+		if client.IgnoreNotFound(err) != nil {
+			return err
+		}
+
+		if _, ok := secret.Data["proxyadmin"]; !ok {
```

Comment:
Is it enough to check just for the existence of the `proxyadmin` key?

What I mean to ask is - are we sure that the pxc-operator won't set this key (with the buggy password) before everest-operator? With the current logic, if the pxc-operator sets a buggy password (before this code executes), then everest-operator will not do anything because the key exists in the secret (and that's all we check)